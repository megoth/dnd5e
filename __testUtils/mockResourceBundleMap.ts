import { ResourceBundleModel } from "../src/models/resourceBundle";
import { getBundleKey, ResourceBundleMap } from "../src/models/app";
import { defaultLocale } from "./mockLanguage";
import mockResourceBundle, { defaultBundle } from "./mockResourceBundle";

export default function mockResourceBundleMap(
  overrides: Partial<ResourceBundleModel> = {}
): ResourceBundleMap {
  const label = overrides.label || defaultBundle;
  const locale = overrides.locale || defaultLocale;
  return {
    [getBundleKey(locale, label)]: mockResourceBundle(overrides),
  };
}
