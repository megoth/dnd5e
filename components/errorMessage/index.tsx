import React from "react";
import { Localized } from "@fluent/react";
import NestedError from "nested-error-stacks";
import { getErrorId } from "../../src/models/error";

interface Props {
  error: NestedError;
}

export const TESTID_ERROR = "error";
export const TESTID_ERROR_TITLE = "error-title";

export default function ErrorMessage({ error }: Props) {
  const { message: errorUrl } = error;
  const errorId = getErrorId(errorUrl);
  return (
    <div data-testid={TESTID_ERROR}>
      <h1 data-testid={TESTID_ERROR_TITLE}>
        <span resource={errorUrl}>
          <Localized id={errorId}>{errorUrl}</Localized>
        </span>
      </h1>
    </div>
  );
}
