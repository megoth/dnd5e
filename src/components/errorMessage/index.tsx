import React from "react";
import NestedError from "nested-error-stacks";

interface Props {
  error: NestedError;
}

export const TESTID_ERROR = "error";
export const TESTID_ERROR_TITLE = "error-title";

export default function ErrorMessage({ error }: Props) {
  // const { message: errorURL } = error;
  return (
    <div data-testid={TESTID_ERROR}>
      <h1 data-testid={TESTID_ERROR_TITLE}>
        {/*{translationURL ? (*/}
        {/*  <Localized id={getTranslationId(translationURL)}>*/}
        {/*    {errorURL}*/}
        {/*  </Localized>*/}
        {/*) : (*/}
        {error.message}
        {/*)}*/}
      </h1>
    </div>
  );
}
