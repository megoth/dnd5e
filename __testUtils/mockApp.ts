import { AppModel, createApp, getBundleKey } from "../src/models/app";
import mockFluentBundle from "./mockFluentBundle";
import { appVocabURL } from "./mockAppIndexDataset";
import mockLanguage, { defaultLocale } from "./mockLanguage";
import mockResourceBundleMap from "./mockResourceBundleMap";
import mockAppIndex from "./mockAppIndex";

export const errorsURL = "https://example.com/global/errors.ttl";
export const faqsURL = "https://example.com/global/faqs.ttl";
export const translationsURL = "https://example.com/global/translations.ttl";
export const localizationsURL =
  "https://example.com/global/translations.en-US.ttl";

export function mockAppPropValue(val, bundleName = "global", locale = "en-US") {
  return bundleName
    ? {
        [getBundleKey(locale, bundleName)]: val,
      }
    : {
        [locale]: val,
      };
}

export default function mockApp(overrides: Partial<AppModel> = {}): AppModel {
  const currentLocale = overrides.currentLocale || defaultLocale;
  return {
    appVocabURL: overrides.appVocabURL || appVocabURL,
    currentLocale,
    fluentBundles: {
      [currentLocale]: mockFluentBundle({}, translationsURL, {
        locale: currentLocale,
      }),
      ...overrides.fluentBundles,
    },
    languages: [mockLanguage(currentLocale), ...(overrides.languages || [])],
    resourceBundles: {
      ...mockResourceBundleMap({
        label: "admin",
        locale: currentLocale,
        data: {
          errors: null,
          faqs: null,
          localizations: null,
          translations: null,
        },
      }),
      ...mockResourceBundleMap({
        locale: currentLocale,
        data: {
          errors: null,
          faqs: null,
          localizations: null,
          translations: null,
        },
      }),
      ...overrides.resourceBundles,
    },
  };
}

export function mockNorwegianApp(overrides: Partial<AppModel> = {}): AppModel {
  const locale = "nb-NO";
  return mockApp({
    currentLocale: locale,
    fluentBundles: {
      [locale]: mockFluentBundle({}, translationsURL, {
        locale,
      }),
    },
    languages: [
      mockLanguage(locale, {
        languageCode: "ðŸ‡³ðŸ‡´",
      }),
    ],
    resourceBundles: {
      ...mockResourceBundleMap({
        locale,
        data: {
          errors: null,
          faqs: null,
          localizations: null,
          translations: null,
        },
      }),
    },
    ...overrides,
  });
}

export function mockEmptyApp(): AppModel {
  return createApp(defaultLocale, appVocabURL, null);
}

export function mockUnloadedApp(): AppModel {
  return createApp(defaultLocale, appVocabURL, mockAppIndex());
}
