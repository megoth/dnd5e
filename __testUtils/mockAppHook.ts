import mockApp from "./mockApp";

export default function mockAppHook(mock: jest.Mock, app = mockApp()) {
  mock.mockReturnValue(app);
}
