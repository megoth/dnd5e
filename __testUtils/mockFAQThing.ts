import { mockThingFrom, setUrl } from "@inrupt/solid-client";
import { rdf } from "rdf-namespaces";
import { getAppTerm } from "../src/models/appIndex";
import { appVocabURL } from "./mockAppIndexDataset";
import { chain } from "../src/utils";
import { faqsURL } from "./mockApp";

export const faqId = "faq";
export const faqURL = `${faqsURL}#${faqId}`;
export const faqLabelURL = "https://example.com/#faqLabel";
export const faqDescriptionURL = "https://example.com/#faqDescription";

export default function mockFAQThing(id = faqId) {
  return chain(
    mockThingFrom(`${faqsURL}#${id}`),
    (t) => setUrl(t, rdf.type, getAppTerm("FAQ", { appVocabURL })),
    (t) => setUrl(t, getAppTerm("faqLabel", { appVocabURL }), faqLabelURL),
    (t) =>
      setUrl(
        t,
        getAppTerm("faqDescription", { appVocabURL }),
        faqDescriptionURL
      )
  );
}
