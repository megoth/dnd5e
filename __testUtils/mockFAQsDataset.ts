import {
  mockSolidDatasetFrom,
  setThing,
  SolidDataset,
} from "@inrupt/solid-client";
import { faqsURL } from "./mockApp";
import mockFAQThing from "./mockFAQThing";

export default function mockFAQsDataset(faqs = [mockFAQThing()]) {
  return faqs.reduce<SolidDataset>(
    (memo, faq) => setThing(memo, faq),
    mockSolidDatasetFrom(faqsURL)
  );
}
