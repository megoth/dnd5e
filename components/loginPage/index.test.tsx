import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as solidUIReactFns from "@inrupt/solid-ui-react";
import useApp from "../../src/hooks/useApp";
import mockAppHook from "../../__testUtils/mockAppHook";
import LoginPage, {
  TESTID_LOGIN_PAGE_BUTTON,
  TESTID_LOGIN_PAGE_IDP_FIELD,
  TESTID_LOGIN_PAGE_PREDEFINED_IDP,
  TESTID_LOGIN_PAGE_REMEMBER_CHECKBOX,
} from "./index";
import mockApp, { appName } from "../../__testUtils/mockApp";
import mockResourceBundleMap from "../../__testUtils/mockResourceBundleMap";
import mockFAQsDataset from "../../__testUtils/mockFAQsDataset";
import mockFAQThing from "../../__testUtils/mockFAQThing";
import { mockUnauthenticatedSession } from "../../__testUtils/mockSession";
import { TESTID_ERROR } from "../errorMessage";
import { getProviders } from "../../src/models/provider";

jest.mock("../../src/hooks/useApp");
const mockedAppHook = useApp as jest.Mock;

const app = mockApp({
  resourceBundles: mockResourceBundleMap({
    data: {
      faqs: mockFAQsDataset([
        mockFAQThing("whyLogInWithSolid"),
        mockFAQThing("whatIsSolid"),
        mockFAQThing("whatIsAPod"),
      ]),
    },
  }),
});

describe("LoginPage", () => {
  const idpUrl = "https://example.com";

  let unauthenticatedSession;
  let mockedSessionHook;

  beforeEach(() => mockAppHook(mockedAppHook, app));
  beforeEach(() => {
    unauthenticatedSession = mockUnauthenticatedSession();
    mockedSessionHook = jest
      .spyOn(solidUIReactFns, "useSession")
      .mockImplementation(() => unauthenticatedSession);
  });

  it("renders", () => {
    const { asFragment } = render(<LoginPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("offers a form to initiate authentication process", () => {
    const { getByTestId } = render(<LoginPage />);
    const idpField = getByTestId(TESTID_LOGIN_PAGE_IDP_FIELD);
    userEvent.type(idpField, `${idpUrl}{enter}`);
    const { login } = unauthenticatedSession;
    expect(login).toHaveBeenCalledWith({
      clientName: appName,
      oidcIssuer: idpUrl,
      redirectUrl: "http://localhost/",
    });
  });

  it("offers the option to remember the IdP for later", () => {
    const { getByTestId } = render(<LoginPage />);
    const rememberCheckbox = getByTestId(TESTID_LOGIN_PAGE_REMEMBER_CHECKBOX);
    userEvent.click(rememberCheckbox);
    const idpField = getByTestId(TESTID_LOGIN_PAGE_IDP_FIELD);
    userEvent.type(idpField, `${idpUrl}{enter}`);
    expect(localStorage.getItem("rememberIdP")).toEqual("true");
    expect(localStorage.getItem("idp")).toEqual(idpUrl);
  });

  it("restores states from localStorage if opted-in", () => {
    localStorage.setItem("rememberIdP", "true");
    localStorage.setItem("idp", idpUrl);
    const { getByTestId } = render(<LoginPage />);
    const rememberCheckbox = getByTestId(TESTID_LOGIN_PAGE_REMEMBER_CHECKBOX);
    expect(rememberCheckbox.getAttribute("checked")).toBeDefined();
    const idpField = getByTestId(TESTID_LOGIN_PAGE_IDP_FIELD);
    expect(idpField.getAttribute("value")).toEqual(idpUrl);
  });

  it("does not remember idp if remember me is not checked", () => {
    localStorage.setItem("rememberIdP", "true");
    localStorage.setItem("idp", idpUrl);
    const { getByTestId } = render(<LoginPage />);
    const rememberCheckbox = getByTestId(TESTID_LOGIN_PAGE_REMEMBER_CHECKBOX);
    userEvent.click(rememberCheckbox);
    const submitButton = getByTestId(TESTID_LOGIN_PAGE_BUTTON);
    userEvent.click(submitButton);
    expect(localStorage.getItem("rememberIdP")).toEqual("false");
    expect(localStorage.getItem("idp")).toBeNull();
  });

  it("renders an error if something goes wrong", () => {
    mockedSessionHook.mockReturnValue(
      mockUnauthenticatedSession({
        login: () => {
          throw new Error();
        },
      })
    );
    const { getByTestId } = render(<LoginPage />);
    const idpField = getByTestId(TESTID_LOGIN_PAGE_IDP_FIELD);
    userEvent.type(idpField, `${idpUrl}{enter}`);
    expect(getByTestId(TESTID_ERROR)).toBeDefined();
  });

  it("renders a list of predefined IdPs", () => {
    const { getAllByTestId } = render(<LoginPage />);
    const predefinedIdpButton = getAllByTestId(
      TESTID_LOGIN_PAGE_PREDEFINED_IDP
    )[0];
    userEvent.click(predefinedIdpButton);
    const { login } = unauthenticatedSession;
    expect(login).toHaveBeenCalledWith({
      clientName: appName,
      oidcIssuer: getProviders()[0].iri,
      redirectUrl: "http://localhost/",
    });
  });
});
