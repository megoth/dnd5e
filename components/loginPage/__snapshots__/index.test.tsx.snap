// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col relative min-h-full"
  >
    <header
      class="mb-2 shadow shadow-md"
    >
      <div
        class="main-container main-container--content pr-0"
      >
        <a
          class="px-4 lg:px-0 font-2xl font-serif self-center text-left focus:outline-none focus:ring-2 focus:ring-red-600"
          href="/"
        >
          [Translation for https://example.com/global/translations.ttl#appName does not exist]
        </a>
      </div>
      <div
        class="bg-gray-100 dark:bg-gray-900"
      >
        <div
          class="main-container main-container--content flex flex-row px-0"
        >
          <button
            class="p-2 px-4 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-red-600"
            data-testid="page-header-left-menu-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="flex-1 text-lg font-semibold font-serif leading-10"
          >
            [Translation for https://example.com/global/translations.ttl#loginPageTitle does not exist]
          </div>
          <button
            class="p-2 px-4 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-red-600"
            data-testid="page-header-right-menu-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 16l-4-4m0 0l4-4m-4 4h14m-5 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h7a3 3 0 013 3v1"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </header>
    <div
      class="main-container main-container--full"
    >
      <div
        class="layout__sidebar layout__sidebar--left layout__sidebar--full"
        data-testid="layout-sub-menu"
      >
        <button
          class="button button--close layout__close-button layout__close-button--left  layout__close-button--full text-right"
          data-testid="sub-menu-nav-close-button"
          type="button"
        >
          <svg
            class="icon"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Â [Translation for https://example.com/global/translations.ttl#close does not exist]
        </button>
        <nav
          class="pages-nav pages-nav--sub-menu"
        >
          <ul
            class="nav-pages__list nav-pages__list--sub-menu"
          >
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/characters"
              >
                [Translation for https://example.com/global/translations.ttl#charactersPageTitle does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/rules"
              >
                [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/faq"
              >
                [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/about"
              >
                [Translation for https://example.com/global/translations.ttl#aboutPageTitle does not exist]
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="layout__main layout__main--full"
      >
        <main
          class="flex-1"
        >
          <div
            class="main-container px-4 lg:px-0 md:grid grid-cols-2 gap-4 max-w-3xl place-items-center"
          >
            <div>
              <div
                class="content leading-normal max-w-prose mx-auto"
                style="hyphens: auto;"
              >
                <h1>
                  [Translation for https://example.com/global/translations.ttl#loginPageTitle does not exist]
                </h1>
                <p>
                  [Translation for https://example.com/global/translations.ttl#authenticationPitch does not exist]
                </p>
              </div>
              <p
                class="font-bold mt-6"
              >
                [Translation for https://example.com/global/translations.ttl#authenticationGuidance does not exist]
              </p>
              <ul
                class="flex flex-col space-y-2 my-2"
              >
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    Inrupt Pod Spaces
                  </button>
                </li>
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    solidcommunity.net
                  </button>
                </li>
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    inrupt.net
                  </button>
                </li>
              </ul>
            </div>
            <form
              class="mt-2 md:mt-0 text-left border p-2 rounded"
            >
              <img
                alt="[Translation for https://example.com/global/translations.ttl#loginImage does not exist]"
                class="mx-auto hidden md:block"
                src="/logos/solid-emblem.svg"
                style="max-width: 200px;"
              />
              <label
                class="label"
                for="idp"
              >
                [Translation for https://example.com/global/translations.ttl#solidIdP does not exist]
                <input
                  class="input"
                  data-testid="login-form-idp-field"
                  id="idp"
                  placeholder="https://"
                  required=""
                  type="url"
                  value=""
                />
              </label>
              <label
                class="label"
                for="rememberIdp"
              >
                <input
                  class="checkbox"
                  data-testid="login-form-remember-checkbox"
                  id="rememberIdp"
                  type="checkbox"
                />
                <span
                  class="label__text"
                >
                  [Translation for https://example.com/global/translations.ttl#rememberIdP does not exist]
                </span>
              </label>
              <button
                class="button w-full"
                data-testid="login-form-button"
                type="submit"
              >
                [Translation for https://example.com/global/translations.ttl#login does not exist]
              </button>
              <div
                class="mt-4"
              >
                <a
                  class="link"
                  href="/signup"
                >
                  [Translation for https://example.com/global/translations.ttl#signupSolidPrompt does not exist]
                </a>
              </div>
            </form>
          </div>
          <aside
            class="main-container main-container--content my-16"
          >
            <div
              class="content leading-normal max-w-prose mx-auto content content--small"
              style="hyphens: auto;"
            >
              <h2
                data-testid="faq-label"
                id="faqLabel"
              >
                [Translation for https://example.com/#faqLabel does not exist]
              </h2>
              <div
                data-testid="faq-description"
              >
                <p>
                  [Translation for https://example.com/#faqDescription does not exist]
                </p>
              </div>
            </div>
            <div
              class="content leading-normal max-w-prose mx-auto content content--small"
              style="hyphens: auto;"
            >
              <h2
                data-testid="faq-label"
                id="faqLabel"
              >
                [Translation for https://example.com/#faqLabel does not exist]
              </h2>
              <div
                data-testid="faq-description"
              >
                <p>
                  [Translation for https://example.com/#faqDescription does not exist]
                </p>
              </div>
            </div>
            <div
              class="content leading-normal max-w-prose mx-auto content content--small"
              style="hyphens: auto;"
            >
              <h2
                data-testid="faq-label"
                id="faqLabel"
              >
                [Translation for https://example.com/#faqLabel does not exist]
              </h2>
              <div
                data-testid="faq-description"
              >
                <p>
                  [Translation for https://example.com/#faqDescription does not exist]
                </p>
              </div>
            </div>
          </aside>
        </main>
        <footer
          class="main-container main-container--content mt-8 text-center sm:text-left xs:flex xs:space-x-4 items-center justify-center sm:justify-start"
        >
          <div
            class="md:flex-grow pb-2 flex space-x-1 justify-center sm:justify-start"
          >
            <a
              class="link"
              href="/about"
            >
              [Translation for https://example.com/global/translations.ttl#aboutPageTitle does not exist]
            </a>
            <a
              class="link"
              href="/faq"
            >
              [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
            </a>
          </div>
          <div
            class="pb-2 flex xs:space-x-2 space-y-2 xs:space-y-0 flex-col text-center xs:flex-row xs:justify-center"
          >
            <button
              class="button"
              data-testid="dark-mode-selector-button"
              type="button"
            >
              <span>
                ðÂ [Translation for https://example.com/global/translations.ttl#darkModeTurnOn does not exist]
              </span>
            </button>
          </div>
        </footer>
      </div>
      <div
        class="layout__sidebar layout__sidebar--right-fake layout__sidebar--full"
      />
      <div
        class="layout__sidebar layout__sidebar--right layout__sidebar--full"
      >
        <div>
          <button
            class="button button--close layout__close-button layout__close-button--right  layout__close-button--full text-left"
            data-testid="session-close-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Â [Translation for https://example.com/global/translations.ttl#close does not exist]
          </button>
          <div
            class="px-2"
          >
            <div
              class="text-center"
            >
              <h2
                class="font-bold text-xl"
              >
                [Translation for https://example.com/global/translations.ttl#recommendedIdPSet does not exist]
              </h2>
              <ul
                class="flex flex-col space-y-2 my-2"
              >
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    Inrupt Pod Spaces
                  </button>
                </li>
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    solidcommunity.net
                  </button>
                </li>
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    inrupt.net
                  </button>
                </li>
              </ul>
              <h2
                class="line-behind font-bold text-xl"
              >
                <span
                  class="line-behind__line"
                />
                <span
                  class="bg-white dark:bg-gray-800 lg:bg-white lg:dark:bg-gray-800"
                >
                  [Translation for https://example.com/global/translations.ttl#or does not exist]
                </span>
              </h2>
              <form
                class="mt-2 md:mt-0 text-left"
              >
                <img
                  alt="[Translation for https://example.com/global/translations.ttl#loginImage does not exist]"
                  class="mx-auto hidden md:block"
                  src="/logos/solid-emblem.svg"
                  style="max-width: 200px;"
                />
                <label
                  class="label"
                  for="idp"
                >
                  [Translation for https://example.com/global/translations.ttl#solidIdP does not exist]
                  <input
                    class="input"
                    data-testid="login-form-idp-field"
                    id="idp"
                    placeholder="https://"
                    required=""
                    type="url"
                    value=""
                  />
                </label>
                <label
                  class="label"
                  for="rememberIdp"
                >
                  <input
                    class="checkbox"
                    data-testid="login-form-remember-checkbox"
                    id="rememberIdp"
                    type="checkbox"
                  />
                  <span
                    class="label__text"
                  >
                    [Translation for https://example.com/global/translations.ttl#rememberIdP does not exist]
                  </span>
                </label>
                <button
                  class="button w-full"
                  data-testid="login-form-button"
                  type="submit"
                >
                  [Translation for https://example.com/global/translations.ttl#login does not exist]
                </button>
                <div
                  class="mt-4"
                >
                  <a
                    class="link"
                    href="/signup"
                  >
                    [Translation for https://example.com/global/translations.ttl#signupSolidPrompt does not exist]
                  </a>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LoginPage renders a warning when authenticated 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col relative min-h-full"
  >
    <header
      class="mb-2 shadow shadow-md"
    >
      <div
        class="main-container main-container--content pr-0"
      >
        <a
          class="px-4 lg:px-0 font-2xl font-serif self-center text-left focus:outline-none focus:ring-2 focus:ring-red-600"
          href="/"
        >
          [Translation for https://example.com/global/translations.ttl#appName does not exist]
        </a>
      </div>
      <div
        class="bg-gray-100 dark:bg-gray-900"
      >
        <div
          class="main-container main-container--content flex flex-row px-0"
        >
          <button
            class="p-2 px-4 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-red-600"
            data-testid="page-header-left-menu-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="flex-1 text-lg font-semibold font-serif leading-10"
          >
            [Translation for https://example.com/global/translations.ttl#loginPageTitle does not exist]
          </div>
          <button
            class="p-2 px-4 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-red-600"
            data-testid="page-header-right-menu-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </header>
    <div
      class="main-container main-container--full"
    >
      <div
        class="layout__sidebar layout__sidebar--left layout__sidebar--full"
        data-testid="layout-sub-menu"
      >
        <button
          class="button button--close layout__close-button layout__close-button--left  layout__close-button--full text-right"
          data-testid="sub-menu-nav-close-button"
          type="button"
        >
          <svg
            class="icon"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Â [Translation for https://example.com/global/translations.ttl#close does not exist]
        </button>
        <nav
          class="pages-nav pages-nav--sub-menu"
        >
          <ul
            class="nav-pages__list nav-pages__list--sub-menu"
          >
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/characters"
              >
                [Translation for https://example.com/global/translations.ttl#charactersPageTitle does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/rules"
              >
                [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/faq"
              >
                [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                data-testid="nav-item-link"
                href="/about"
              >
                [Translation for https://example.com/global/translations.ttl#aboutPageTitle does not exist]
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="layout__main layout__main--full"
      >
        <main
          class="flex-1"
        >
          <div
            class="main-container main-container--content"
            data-testid="logged-in-already-warning"
          >
            <div
              class="bg-yellow-200 text-yellow-900 border-yellow-400 border px-2 py-1 rounded-sm mb-2"
            >
              [Translation for https://example.com/global/translations.ttl#LoggedInAlready does not exist]
            </div>
          </div>
          <div
            class="main-container px-4 lg:px-0 md:grid grid-cols-2 gap-4 max-w-3xl place-items-center"
          >
            <div>
              <div
                class="content leading-normal max-w-prose mx-auto"
                style="hyphens: auto;"
              >
                <h1>
                  [Translation for https://example.com/global/translations.ttl#loginPageTitle does not exist]
                </h1>
                <p>
                  [Translation for https://example.com/global/translations.ttl#authenticationPitch does not exist]
                </p>
              </div>
              <p
                class="font-bold mt-6"
              >
                [Translation for https://example.com/global/translations.ttl#authenticationGuidance does not exist]
              </p>
              <ul
                class="flex flex-col space-y-2 my-2"
              >
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    Inrupt Pod Spaces
                  </button>
                </li>
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    solidcommunity.net
                  </button>
                </li>
                <li>
                  <button
                    class="button button--solid w-full"
                    data-testid="login-button"
                    type="button"
                  >
                    inrupt.net
                  </button>
                </li>
              </ul>
            </div>
            <form
              class="mt-2 md:mt-0 text-left border p-2 rounded"
            >
              <img
                alt="[Translation for https://example.com/global/translations.ttl#loginImage does not exist]"
                class="mx-auto hidden md:block"
                src="/logos/solid-emblem.svg"
                style="max-width: 200px;"
              />
              <label
                class="label"
                for="idp"
              >
                [Translation for https://example.com/global/translations.ttl#solidIdP does not exist]
                <input
                  class="input"
                  data-testid="login-form-idp-field"
                  id="idp"
                  placeholder="https://"
                  required=""
                  type="url"
                  value=""
                />
              </label>
              <label
                class="label"
                for="rememberIdp"
              >
                <input
                  class="checkbox"
                  data-testid="login-form-remember-checkbox"
                  id="rememberIdp"
                  type="checkbox"
                />
                <span
                  class="label__text"
                >
                  [Translation for https://example.com/global/translations.ttl#rememberIdP does not exist]
                </span>
              </label>
              <button
                class="button w-full"
                data-testid="login-form-button"
                type="submit"
              >
                [Translation for https://example.com/global/translations.ttl#login does not exist]
              </button>
              <div
                class="mt-4"
              >
                <a
                  class="link"
                  href="/signup"
                >
                  [Translation for https://example.com/global/translations.ttl#signupSolidPrompt does not exist]
                </a>
              </div>
            </form>
          </div>
          <aside
            class="main-container main-container--content my-16"
          >
            <div
              class="content leading-normal max-w-prose mx-auto content content--small"
              style="hyphens: auto;"
            >
              <h2
                data-testid="faq-label"
                id="faqLabel"
              >
                [Translation for https://example.com/#faqLabel does not exist]
              </h2>
              <div
                data-testid="faq-description"
              >
                <p>
                  [Translation for https://example.com/#faqDescription does not exist]
                </p>
              </div>
            </div>
            <div
              class="content leading-normal max-w-prose mx-auto content content--small"
              style="hyphens: auto;"
            >
              <h2
                data-testid="faq-label"
                id="faqLabel"
              >
                [Translation for https://example.com/#faqLabel does not exist]
              </h2>
              <div
                data-testid="faq-description"
              >
                <p>
                  [Translation for https://example.com/#faqDescription does not exist]
                </p>
              </div>
            </div>
            <div
              class="content leading-normal max-w-prose mx-auto content content--small"
              style="hyphens: auto;"
            >
              <h2
                data-testid="faq-label"
                id="faqLabel"
              >
                [Translation for https://example.com/#faqLabel does not exist]
              </h2>
              <div
                data-testid="faq-description"
              >
                <p>
                  [Translation for https://example.com/#faqDescription does not exist]
                </p>
              </div>
            </div>
          </aside>
        </main>
        <footer
          class="main-container main-container--content mt-8 text-center sm:text-left xs:flex xs:space-x-4 items-center justify-center sm:justify-start"
        >
          <div
            class="md:flex-grow pb-2 flex space-x-1 justify-center sm:justify-start"
          >
            <a
              class="link"
              href="/about"
            >
              [Translation for https://example.com/global/translations.ttl#aboutPageTitle does not exist]
            </a>
            <a
              class="link"
              href="/faq"
            >
              [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
            </a>
          </div>
          <div
            class="pb-2 flex xs:space-x-2 space-y-2 xs:space-y-0 flex-col text-center xs:flex-row xs:justify-center"
          >
            <button
              class="button"
              data-testid="dark-mode-selector-button"
              type="button"
            >
              <span>
                ðÂ [Translation for https://example.com/global/translations.ttl#darkModeTurnOn does not exist]
              </span>
            </button>
          </div>
        </footer>
      </div>
      <div
        class="layout__sidebar layout__sidebar--right-fake layout__sidebar--full"
      />
      <div
        class="layout__sidebar layout__sidebar--right layout__sidebar--full"
      >
        <div>
          <button
            class="button button--close layout__close-button layout__close-button--right  layout__close-button--full text-left"
            data-testid="session-close-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Â [Translation for https://example.com/global/translations.ttl#close does not exist]
          </button>
          <div
            class="px-2"
          >
            <p>
              [Translation for https://example.com/global/translations.ttl#loggedIn does not exist]
            </p>
            <button
              class="button w-full"
              data-testid="authenticated-logout-button"
              type="button"
            >
              [Translation for https://example.com/global/translations.ttl#logOut does not exist]
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
